addon/lazy-router.js

```diff
@@ -39,9 +39,29 @@ export default Ember.Router.extend({
       return handler;
     };
   },
+  _getContainer: function() {
+    var hasGetOwner = typeof getOwner === "function";
+    var container = hasGetOwner? getOwner(this) : this.container;
```

Comment:
@MiguelMadero , Yes, you are correct. `getOwner(this)` has a property called `base` and the factory has to be registered in this base.
I've made those changes and the correct route from the package is not from the container.

I think, this branch is good to go now.